### ISSUE TITLE
time constraint

### ISSUE THREAD

**Issue by T-King** [CSCE360025/Assignment6#4]
I need to start working on other homework. The due time is 9 pm I believe. I will check back in around 8 to see if anything else needs to be done. I hope that doesn't inconvenience you too much.

**Comment by T-King** [comment]
The time has been extended to 11:30 on blackboard. I really need to work on something else now. Really wish the other 2 would do anything.

**Comment by krustchinsky** [comment]
I'll get it to a semi-working state and then turn it in. I'll let the TA know they didn't contribute.

**Comment by T-King** [comment]
Thank you. It wasn't my intention to burden you with this, but I got behind.

**Comment by tlemaster92** [comment]
Sorry I was under the impression y'all didn't want help. I don't get home from work till 3 tonight but I can help out when I do get home. Just let me know what I need to do.

**Comment by T-King** [comment]
The assignment has been here for a week, but I don't know if you were even in class when Sweany assigned it. And it doesn't really matter what anybody wants. We all have to participate in some way. Regardless, the assignment is due at 11:30 so there's nothing to do about it for this assignment.

**Comment by tlemaster92** [comment]
Don't really know how me being not being in class has anything to do with yall ... But regardless, sorry about not helping out so far, been over my head this semester. I'll get more involved the second half of this semester.

**Comment by ScotuFlash** [comment]
Yeah.. Same here. Just got home from work right now actually. If anything I feel like I would have messed up whatever y'all have going. If it makes any difference I can tackle the next minor assignment all on my own. I know I can do the code, but I am no where close to the level of both Troy and Jacob when it comes to coding. I know that's no excuse, but I feel I would cause more bugs than fixes. What I am semi-good at is note taking and test taking. I regret not sending you guys the outline of notes I had for the past exam. I studied it for about 4 hours the night before and ended up 1 point away from getting a B. Again, I know this doesn't make up for not coding.

By the way, Jacob, mind if I ask what you got on the exam?

**Comment by tlemaster92** [comment]
So I really think for the next assignment we need to actually divvy up the work and assign parts to everyone. Cause I feel the same way as Andrew. I know I'm not a good coder so I feel hesitent about helping. I havent helped because im pretty sure when im involved its gunna take longer than necissary becouse yall will have to go behind me and fix my shit or take double the time explaing everything to me. Another option is we can schedule a time to all get together and work it as a team so we all can give input and know what's going on.

**Comment by ScotuFlash** [comment]
To be fair:
No one is making any effort to cultivate a cooperative environment.
And I understand at the beginning everyone but Jacob didn't know how to use Github.
I have not seen Jacob or Taylor in class since the day of the exam, although I did miss last Friday.
Not showing up to class and not learning from lecture is going to hurt us on the exam that we all need to receive a 50+ on (which by the way is weighted WAY higher than ALL of the minor assignments grouped together; 30% of grade - exam versus 10% of grade - ALL minor assignments).
No one has tried scheduling a time for us to cooperate and see who can do what, Jacob just does all the work, yes, just Jacob.
Sorry Troy, but to be fair, you only contributed 13 lines of code on this assignment and Jacob did 644, it would hardly be fair for you to receive as much credit as Jacob; proof attached in the image.

![assignment 6](https://cloud.githubusercontent.com/assets/10774706/6912514/0f3e5062-d738-11e4-9574-7a55ab30dd7a.JPG)

**Comment by krustchinsky** [comment]
Andrew, if we are being honest let's be entirely honest. We are all adults, the material is available to everyone, and a person's inability to use their resources to be an asset to the team is ultimately not the failing of the team but that person. I've made many efforts to keep everyone on the same page, no E-mails are ever responded to, and no effort is made in class or outside of class to communicate as a group outside of my efforts. The TA has been aware of this since the start of the semester, and it is well documented.

I have not been in class because I have had acute bronchitis since the start of spring break, I will be in class on Wednesday. This however is not excuse for any other team member for a lack of participation that is again ... well documented. We all are in college full time, we all have jobs, and we all have responsibilities. Group work is a component of this class that is primarily focused on homework assignments, the major project and rarely in-class assignments. To blame any lack of attendance on any person's part is dishonest in examining the lack of participation in homework. Also Andrew, something our professor has made abundantly clear since the start of the semester is that he ultimately decides how the final grades for assignments are calculated. It has been said my many former students of his that the exams he gives are always abysmal, and ultimately are curved heavily.

If Github really is the issue, it has never been properly communicated and addressed. If you wish to see change, make the change. I will not accept "we don't know how to use it" as an excuse for the failings of an individuals participation, because the work the is contained in the homework assignments is ultimately much more difficult then git and Github. As computer science students this shouldn't require much effort to learn, I've sent numerous tutorials, explained twice in-class, and if those sessions were not enough that should have been vocalized and it would have been addressed. 

On the question of Troy's participation up to this point, I can admit it was rocky getting started but one key differentiator in his case compared to both of yours if that he is actually showing some inkling of initiative/effort. Your metrics are inaccurate Andrew as Troy made the majority of the commits for Assignment #6 over the course of the past week, and I would venture to guess his contributions were greater than or equal to my own on that particular assignment.

If we really are choosing now to get upset about a lack of participation, then the issue becomes very clear. It is not one's in-abilty (obvious by the swift responses tonight from an otherwise quiet group) but instead ones' laziness. The wonderful thing about Github is that it doubles as an accountability tool (I made you all aware of this) so that if chaos erupts in the group (as it has) it can easily be crushed by sheer metrics.

**Comment by ScotuFlash** [comment]
So now that that's out of the way:
Let's all show up to class, the one time we can all be in the same place at the same time, and explain each assignment to each other (part by part, teaching implementation, who's doing what, etc) instead of blindly not seeing group members and randomly receiving emails about code you've written that I might not understand.

**Comment by krustchinsky** [comment]
Class time is not group project time, this was made clear by the TA and Sweany at the start of the semester. Again, lack of attendance by any person is not what I feel is the primary issue. We are all computer science students, there are literally millions of ways we can communicate without being in front of one another. If you have trouble understanding code I am more than willing to discuss it at any length, in any medium you see fit. I respond to E-mails, issues, text messages, phone calls, etc. All of this information was given out, it has never been used except by Troy and myself.

We can all agree to come to class, but let's be honest, the tiny time-frame at the end of class (following lectures) will never be sufficient time to discuss any assignment we get.

**Comment by ScotuFlash** [comment]
In the tiny time frame after class we can divvy up the work as Taylor said. That's what all the other groups do. There's a big difference between each doing our assigned part and the way we've been going at it.

**Comment by krustchinsky** [comment]
I'm fine with this going forward, but this is still not a reasonable excuse for previous failings. I can't believe that neither of your contributed for the past few months because of anything outside of a lack of effort.

On Wednesday we can take a few moments after class to come up with a better strategy for communication or collaboration is need be.

**Comment by T-King** [comment]
I'll be in class tomorrow. I had a cache class idea I wanted to share anyways.
